src/shmem_synchronization.h

```diff
@@ -42,6 +43,7 @@ shmem_internal_fence(shmem_ctx_t ctx)
     if (0 != ret) { RAISE_ERROR(ret); }
 
     shmem_internal_membar_store();
+    shmem_transport_syncmem();
```

Comment:
I think this call may not be needed since fence doesn't provide any guarantees about what's visible in memory.